Added `yarn` cache to CI workflows

oscard0m opened the pull request:
## Description

Add `cache` to workflows using `actions/setup-node`

## Context

`setup-node` GitHub Action just released a new option to add cache to steps using it.

You can find the details here: https://github.blog/changelog/2021-07-02-github-actions-setup-node-now-supports-dependency-caching/

Re-generated PR from https://github.com/TryGhost/Ghost/pull/13597

---

🤖 This PR has been generated automatically by [this octoherd script](https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action), feel free to run it in your GitHub user/org repositories! 💪🏾

oscard0m committed:
ci(workflow): add cache 'yarn' to workflows

oscard0m commented:
@daniellockyer I did a cleanup of my forks and I had to re-generate the fork, PR's and so (not sure if there was a way to link everything). Closing old PR

daniellockyer commented:
@oscard0m Thanks for the updated PR! It looks good to me the the `setup-node` actions seems to fail upon the job finishing - do you know why?

vikaspotluri123 commented:
@daniellockyer I worked around this issue in a different project by creating the yarn cache directory (`mkdir -p $(yarn cache dir)`).
But... it seems like that means the package cache is empty 😕

daniellockyer commented:
https://github.com/actions/setup-node/issues/317#issuecomment-929694556 🤯

oscard0m commented:
> [actions/setup-node#317 (comment)](https://github.com/actions/setup-node/issues/317#issuecomment-929694556) 🤯

OMG! Do you plan to block this until [that issue](https://github.com/actions/setup-node/issues/317#issuecomment-929694556) is addressed?

daniellockyer commented:
@oscard0m I've implemented a [workaround](https://github.com/TryGhost/Ghost/commit/aec14e5cf34bbb96720e303c32e1cae3fc884169) so we didn't have to disable `FORCE_COLOR` for the whole workflow. I've tested with your PR on top and caching seems to work!

If you'd just like to give it a quick rebase + then we should be good to merge 🙂

oscard0m commented:
> @oscard0m I've implemented a [workaround](https://github.com/TryGhost/Ghost/commit/aec14e5cf34bbb96720e303c32e1cae3fc884169) so we didn't have to disable `FORCE_COLOR` for the whole workflow. I've tested with your PR on top and caching seems to work!
> 
> If you'd just like to give it a quick rebase + then we should be good to merge 🙂

Thanks @daniellockyer . Rebased pushed. Let's see what the CI says now 🍿

daniellockyer commented:
@oscard0m Thanks for the PR! 🕺🏻
